pkgs/tools/networking/openssh/common.nix

```diff
@@ -182,4 +183,9 @@ stdenv.mkDerivation {
     maintainers = (extraMeta.maintainers or []) ++ (with maintainers; [ eelco aneeshusa ]);
     mainProgram = "ssh";
   } // extraMeta;
-}
+} // lib.optionalAttrs isNixos {
+  # openssh calls passwd to allow the user to reset an expired password, but nixos
+  # doesn't ship it at /usr/bin/passwd.
+  PATH_PASSWD_PROG = "/run/wrappers/bin/passwd";
```

Comment:
I always find it a bit cleaner to not use optionalAttrs for such things. We can probably place this inside of env and use optionalAttrs there. What do you think?

Reply:
Yeah, that's much nicer. Wasn't aware of `env`!